localstack-core/localstack/services/secretsmanager/provider.py

```diff
@@ -729,11 +729,15 @@ def backend_rotate_secret(
     if not self._is_valid_identifier(secret_id):
         raise SecretNotFoundException()
 
-    if self.secrets[secret_id].is_deleted():
+    secret = self.secrets[secret_id]
+    if secret.is_deleted():
         raise InvalidRequestException(
             "An error occurred (InvalidRequestException) when calling the RotateSecret operation: You tried to \
             perform the operation on a secret that's currently marked deleted."
         )
+    # Resolve rotation_lambda_arn and fallback to previous value if its missing
+    # from the current request
+    rotation_lambda_arn = rotation_lambda_arn or secret.rotation_lambda_arn
```

Comment:
What if the secret is being rotated first time and we haven't specified the `rotation_lambda_arn`?

Reply:
Actually, this was an existing behavior even before I introduce my fix. In case for the first time rotation we did not specify `rotation_lambda_arn` then it will raise error, check this code here 
```

    try:
        lm_client = connect_to(region_name=self.region_name).lambda_
        lm_client.get_function(FunctionName=rotation_lambda_arn)
    except Exception:
        raise ResourceNotFoundException("Lambda does not exist or could not be accessed")
```

This will raise error whenever we invoked it from the cli

`An error occurred (ResourceNotFoundException) when calling the RotateSecret operation: Lambda does not exist or could not be accessed`

Reply:
What error message does AWS return in this case? Does it match the error we raise in this circumstance? I suspect not.

Reply:
you are right. I will update the code because I need to add a check earlier to raise an error that match AWS one

Reply:
The error mainly look like this from AWS

```
        "Error": {
          "Code": "InvalidRequestException",
          "Message": "No Lambda rotation function ARN is associated with this secret."
        },
```